src/main/kotlin/minesweeper/entity/Cell.kt

```diff
@@ -0,0 +1,13 @@
+package minesweeper.entity
+
+sealed class Cell {
+    abstract val coordinate: Coordinate
+
+    fun matches(coordinate: Coordinate): Boolean {
+        return this.coordinate == coordinate
+    }
+
+    data class Mine(override val coordinate: Coordinate) : Cell()
+
+    data class Empty(override val coordinate: Coordinate) : Cell()
+}
```

Comment:
[상속보다는 조합의 방식을 활용](https://tecoble.techcourse.co.kr/post/2020-05-18-inheritance-vs-composition/)해보면 어떨까요? 😃

```suggestion
sealed class Cell(val coordinate: Coordinate) {

    fun matches(coordinate: Coordinate): Boolean {
        return this.coordinate == coordinate
    }

    class Mine(coordinate: Coordinate) : Cell(coordinate)

    class Empty(coordinate: Coordinate) : Cell(coordinate)
}
```